docs/rule/sort-invocations.md

```diff
@@ -0,0 +1,97 @@
+# sort-invocations
+
+🔧 The `--fix` option on the command line can automatically fix some of the problems reported by this rule.
+
+## Why use it?
+
+The rule helps you standardize templates:
+
+- Component invocations
+- Helper invocations
+- Modifier invocations
+
+By sorting things that are order-independent, you can more easily refactor code. In addition, sorting removes style differences, so you can review another person's code more effectively.
+
+> [!TIP]
+>
+> The `--fix` option for this rule doesn't preserve formatting. To preserve formatting, you can use [`ember-template-lint-plugin-prettier`](https://github.com/ember-template-lint/ember-template-lint-plugin-prettier) and [`prettier-plugin-ember-template-tag`](https://github.com/ember-tooling/prettier-plugin-ember-template-tag) to format `*.hbs` and `*.{gjs,gts}`, respectively.
+
+## Examples
+
+### Components
+
+When invoking a component, list things in the following order:
+
+1. Arguments
+2. Attributes
+3. Modifiers
+4. Splattributes
+
+The order clearly shows how the component is customized more and more. Things are alphabetized within each group.
+
+```hbs
+<Ui::Button
+  @label="Submit form"
+  @type="submit"
+  data-test-button
+  {{on "click" this.doSomething}}
+  ...attributes
+/>
+```
+
+### Helpers
+
+When invoking a helper, list the named arguments in alphabetical order.
+
+```hbs
+{{t
+  "my-component.description"
+  installedOn=this.installationDate
+  packageName="@ember/source"
+  packageVersion="6.0.0"
+}}
+```
+
+### Modifiers
+
+Similarly to helpers, list the named arguments in alphabetical order.
+
+## Limitations
+
+It's intended that there are no options for sorting. Alphabetical sort is the simplest for everyone to understand and to apply across different projects. It's also the easiest to maintain.
+
+To better meet your needs, consider creating a plugin for `ember-template-lint`.
+
+## Known issues
+
+1\. If you passed an empty string as an argument's value, it has been replaced with the placeholder text `<EMPTY STRING>`. Do a find-and-replace-all to restore the value.
```

Comment:
I'm suggesting to use `@decription={{""}}` as safe fallback to avoid `not-reading-documentation` error.

Reply:
Ah, interesting.

I checked that the syntax `{{""}}` will cause [`no-unnecessary-curly-strings`](https://github.com/ember-template-lint/ember-template-lint/blob/v7.5.0-ember-template-lint/docs/rule/no-unnecessary-curly-strings.md) to fail. I can update the code to insert `{{""}}` instead, but we'll need to rely on the autofix from `no-unnecessary-curly-strings` (i.e. we create a dependency between two rules).

Reply:
@ijlee2 how about `{{concat ""}}` ?

Reply:
Code that will run into this issue originally looks something like,

```hbs
<MyComponent @description="" />
```

So I think the better option was to let this rule change the syntax to 

```hbs
<MyComponent @description={{""}} />
```

then allow `no-unnecessary-curly-strings`' autofix to change the syntax back to

```hbs
<MyComponent @description="" />
```

If we add `{{concat ""}}`, this would change the code more drastically. Other developers, who didn't run `ember-template-lint` may be confused as to why the `{{concat}}` helper is used to add an empty string.